portal/models/user.py

```diff
@@ -1192,6 +1192,21 @@ def staff_html(self):
             return '<div>' + '</div><div>'.join(
                 [ui.staff_html for ui in uis]) + '</div>'
 
+    def report_html(self):
+        """Helper used from templates to display all patient report information
+
+        Combines the intervention links from staff_html() with the patient's
+        list of UserDocuments of type PatientReport (including download links)
+
+        """
+        html = self.staff_html()
+        for doc in self.documents.filter_by(document_type='PatientReport'):
```

Comment:
Not crazy about mixing html and backend code.

Makes sense to provide a filter accessor, but I'd rather it called from the template and marked up there.

Also, use `url_for` rather than hardcoding request paths.